common/archiver/provider/provider.go

```diff
@@ -25,20 +25,27 @@ import (
 
 	"github.com/uber/cadence/common/archiver"
 	"github.com/uber/cadence/common/archiver/filestore"
-	"github.com/uber/cadence/common/log/tag"
 )
 
 var (
 	// ErrUnknownScheme is the error for unknown archiver scheme
 	ErrUnknownScheme = errors.New("unknown archiver scheme")
+	// ErrEmptyBootStrapContainer is the error for empty bootstrap container
+	ErrEmptyBootStrapContainer = errors.New("empty bootstrap container")
+	// ErrArchiverNotExist is the error when the Archiver requested doesn't not exist yet
+	ErrArchiverNotExist = errors.New("archiver does not exist yet, please use CreateArchiver method")
 )
 
 type (
 	// ArchiverProvider returns history or visibility archiver based on the scheme and serviceName.
-	// The archiver for each combination of scheme and serviceName will be created only once and cached.
+	// Create() method will create a new archiver (overwritting the existing one) and cache it.
+	// Get() method will find the existing archiver for a given scheme and serviceName pair and return it.
+	// If that archiver does not exist, an ErrArchiverNotExist error will be returned.
 	ArchiverProvider interface {
-		GetHistoryArchiver(scheme string, serviceName string, tags ...tag.Tag) (archiver.HistoryArchiver, error)
-		GetVisibilityArchiver(scheme string, serviceName string, tags ...tag.Tag) (archiver.VisibilityArchiver, error)
+		CreateHistoryArchiver(scheme string, serviceName string, container *archiver.HistoryBootstrapContainer) (archiver.HistoryArchiver, error)
```

Comment:
I will change the provider interface as we discussed, in the domain config diff. No need to review this part now.